### PR TITLE
[SYCL] Destroy event_impl dependencies in command class destructor

### PR DESCRIPTION
**Pull request by maximdimakov**
Move destroying of dependencies to command destructor.
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

**Review by steffenlarsen — APPROVED** [review]
LGTM!

**Comment by maximdimakov** [comment]
It seems that pre-commit fail are not related to this patch. Many of last PRs have the same fails

File: sycl/source/detail/scheduler/graph_builder.cpp [hunk 1]
```diff
@@ -129,7 +129,6 @@ static void handleVisitedNodes(std::vector<Command *> &Visited) {
   for (Command *Cmd : Visited) {
     if (Cmd->MMarks.MToBeDeleted) {
       Cmd->getEvent()->setCommand(nullptr);
-      Cmd->getEvent()->cleanupDependencyEvents();
```
Comment (bader) [inline sycl/source/detail/scheduler/graph_builder.cpp:132]: It looks like you removed all uses of `cleanupDependencyEvents`. Can we remove the definition of this function?

Comment (maximdimakov) [reply sycl/source/detail/scheduler/graph_builder.cpp:132]: This function is used in Command::~Command()

Comment (bader) [reply sycl/source/detail/scheduler/graph_builder.cpp:132]: https://github.com/intel/llvm/blob/sycl/sycl/source/detail/scheduler/commands.hpp#L190
I can't find it. Can you point to the exact location in the source code?

Comment (maximdimakov) [reply sycl/source/detail/scheduler/graph_builder.cpp:132]: This PR changes ~Command and this function is called there
https://github.com/intel/llvm/pull/5401/files#diff-5e76d6e04c052ad4472e6884f7f9756e9696fe1f02ae959ed31a327457c33f9aL190

**Pull request merged**